### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by muneeb3123**
This pr includes:

- Use preprocessed HTML file with embedded Ruby code.

- Use layouts and templates for shared content.

- Implement the design from the sneak peek wireframes.

- Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,5 +1,11 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
```
Comment (BregornOriginal) [inline app/controllers/posts_controller.rb:2]: - [ ] To follow the different wireframes, we can see that the home page `http://localhost:3000/` shows a list of users, which means this page is `users#index`. Kindly add this to your `config/routes` to match with the wireframe routes

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1 +1,16 @@
-<h1>This page shows all posts by single user</h1>
\ No newline at end of file
+<main>
+    <div class="single-post-container">
```
Comment (BregornOriginal) [inline app/views/posts/show.html.erb:2]: - [ ] Kindly add your styling to this view to match the requirements on the wireframe provided

Your design (without styles)

image

Wireframe

image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,31 @@
-<h1>This page shows all posts</h1>
\ No newline at end of file
+<main>
+    <div class="posts-main-container">
```
Comment (BregornOriginal) [inline app/views/posts/index.html.erb:2]: - [ ] Kindly add your styling to this view to match the requirements on the wireframe provided

Your design (without styles)

image

Wireframe

image

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -1 +1,17 @@
-<h1>This page shows all users</h1>
\ No newline at end of file
+<main>
+    <div class="users-container" >
```
Comment (BregornOriginal) [inline app/views/users/index.html.erb:2]: - [ ] Kindly add your styling to this view to match the requirements on the wireframe provided

Your design (without styles)

image

Wireframe

image

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1 +1,35 @@
-<h1>This page shows single user by specific id</h1>
\ No newline at end of file
+<main>
+    <div class="single-user-container">
```
Comment (BregornOriginal) [inline app/views/users/show.html.erb:2]: - [ ] Kindly add your styling to this view to match the requirements on the wireframe provided

Your design (without styles)

image

Wireframe

image

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @muneeb3123 

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

hand-sketching-stair-slogan-one-step-closer-chalkboard-79948193

## Highlights

✅ Linters are passing
✅ Good commit history
✅ README looks professional


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @muneeb3123   

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Rails  features ✔️ 
- [x] Kudus on meeting the requirements suggested b the last reviewer ✔️


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**